modules/websocket_streaming_server_module/src/websocket_streaming_server_impl.cpp

```diff
@@ -3,49 +3,98 @@
     websocketStreamingServer.start();
 }
 
-PropertyObjectPtr WebsocketStreamingServerImpl::createDefaultConfig()
+void WebsocketStreamingServerImpl::populateDefaultConfigFromProvider(const ContextPtr& context, const PropertyObjectPtr& config)
+{
+    if (!context.assigned())
+        return;
+    if (!config.assigned())
+        return;
+
+    LoggerComponentPtr loggerComponent;
+
+    auto options = context.getModuleOptions("StreamingLtServer");
+    for (const auto& [key, value] : options)
+    {
+        if (config.hasProperty(key))
+        {
+            ErrCode err = config->setPropertyValue(key, value);
+            if (err != OPENDAQ_SUCCESS)
+            {
+                loggerComponent = context.getLogger().getOrAddComponent("ConfigProvider/Modules/StreamingLtServer");
+                LOG_W("Ignoring property \"{}\". Using default value \"{}\"", key, config.getPropertyValue(key));
+            }
+        }
+        else
+        {
+            auto property = ObjectPropertyBuilder(key, value).setValueType(value.getCoreType()).build();
+            config.addProperty(property);
+        }
+    }
+}
+
+PropertyObjectPtr WebsocketStreamingServerImpl::createDefaultConfig(const ContextPtr& context)
 {
     constexpr Int minPortValue = 0;
     constexpr Int maxPortValue = 65535;
 
     auto defaultConfig = PropertyObject();
 
+    defaultConfig.addProperty(StringProperty("Name", "OpenDAQ_Server"));
+    defaultConfig.addProperty(StringProperty("Manufacturer", "openDAQ"));
+    defaultConfig.addProperty(StringProperty("Model", ""));
+    defaultConfig.addProperty(StringProperty("SerialNumber", ""));
+    defaultConfig.addProperty(BoolProperty("ServiceDiscoverable", false));
+
     const auto websocketPortProp =
-        IntPropertyBuilder("WebsocketStreamingPort", 7414).setMinValue(minPortValue).setMaxValue(maxPortValue).build();
+        IntPropertyBuilder("Port", 7414).setMinValue(minPortValue).setMaxValue(maxPortValue).build();
     defaultConfig.addProperty(websocketPortProp);
 
     const auto websocketControlPortProp =
         IntPropertyBuilder("WebsocketControlPort", 7438).setMinValue(minPortValue).setMaxValue(maxPortValue).build();
     defaultConfig.addProperty(websocketControlPortProp);
 
+    const auto websocketServiceProp = StringPropertyBuilder("ServiceName", "_streaming-lt._tcp.local.")
+        .setReadOnly(true)
+        .build();
+    defaultConfig.addProperty(websocketServiceProp);
+
+    const auto serviceCapProp = StringPropertyBuilder("ServiceCap", "LT")
+        .setReadOnly(true)
+        .build();
+    defaultConfig.addProperty(serviceCapProp);
+
+    defaultConfig.addProperty(StringProperty("ServicePath", "/"));
+
+    populateDefaultConfigFromProvider(context, defaultConfig);
     return defaultConfig;
 }
 
-ServerTypePtr WebsocketStreamingServerImpl::createType()
+ServerTypePtr WebsocketStreamingServerImpl::createType(const ContextPtr& context)
 {
     return ServerType(
         "openDAQ LT Streaming",
```

Comment:
I think we should probably fix these IDs to match the clients. Opinion? We use snake case for native config, we use an ID with spaces here... We should clean these up, but maintain backwards compatiblity.